SU-HSDO/suhumsci
# STN-656: Heading Links Utility Class to add Color

# [STN-656](https://sparkbox.atlassian.net/browse/STN-656) READY FOR REVIEW

## Summary
Refinement of heading color.  Clients that are now asking for more distinct heading styles for links (especially for People). We want to keep the default of heading color as black and add a utility class to change the heading color to the existing link color set in the theme.

The new utility class created for this work is `hb-heading-link-color`.

This utility class has been added to the utility class documentation and to the color pairing spreadsheet.

## Need Review By (Date)
2/15

## Urgency
low

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. Test the newly created `hb-heading-link-color` utility class when applied to the title field in a view, to a view, and to layout builder. Use the testing list below as a guide.
3. Confirm the utility class works as expected in both the Colorful and Traditional themes.
4. Confirm the utility class has no effect when the dark inversion toggle is activated via the theme setting.

### Testing

Utility Class Application Use Cases
- [x] View (hb-heading-link-color)
- [x] View (hb-heading-link-color hb-raised-cards)
- [x] Field (hb-heading-link-color)
- [x] Field (hb-heading-link-color) + View (hb-raised-cards)
- [x] Field (hb-heading-link-color) + Layout Builder (hb-raised-cards)
- [x] Layout Builder (hb-heading-link-color)
- [x] Layout Builder (hb-heading-link-color hb-raised-cards)

The utility class of `hb-heading-link-color` should not be used alongside `hb-dark-pattern` as the color contrast between the two will not pass accessibility.

Themes
- [x] Traditional
- [x] Colorful

Theme Settings
- [x] Dark Inversion Toggle
* Note that if `hb-heading-link-color` is applied via the view or in layout builder, the specificity of the dark inversion toggle class takes precedence, therefore there is not an issue with color contrast / accessibility.

Review the color pairings on Traditional

Review the color pairings on Colorful
